oio/xcute/common/backend.py

```diff
@@ -278,16 +281,19 @@ class XcuteBackend(RedisConnection):
             return redis.error_reply('no_job');
         end;
 
-        local total_tasks_sent = redis.call(
-            'HINCRBY', info_key,
-            'tasks.sent', tasks_sent_length);
-        for _, task_id in ipairs(tasks_sent) do
-            redis.call('SADD', 'xcute:tasks:running:' .. job_id, task_id);
-        end;
+        local nb_tasks_sent = 0;
         if tasks_sent_length > 0 then
+            nb_tasks_sent = redis.call(
+                'SADD', 'xcute:tasks:running:' .. job_id, unpack(tasks_sent));
+            if nb_tasks_sent ~= tasks_sent_length then
+                return redis.error_reply('tasks_already_sent');
```

Comment:
isn't it broken because we don't update the number of tasks sent by `nb_tasks_sent`?